frontend/packages/console-app/src/components/network-policies/network-policy-model.ts

```diff
@@ -136,11 +136,12 @@ const ruleToK8s = (
 export const networkPolicyToK8sResource = (
   from: NetworkPolicy,
 ): NetworkPolicyKind | ConversionError => {
-  const podSelector = selectorToK8s(from.podSelector, null);
+  const podSelector = selectorToK8s(from.podSelector, undefined);
   if (isError(podSelector)) {
     return podSelector;
   }
 
+  const policyTypes: string[] = [];
   const res: NetworkPolicyKind = {
```

Comment:
Hmmm, did TypeScript recommend more typing? Because I would think you are good with the previous design. Which is a tad bit more future proof and I'm not sure you're adding much here as `NetworkPolicyKind` already types `policyTypes`.

Reply:
Yes I had to do that to fix a compilation error. Problem was line 157 where `res.spec.policyTypes.push(networkPolicyTypeIngress);` triggered `res.spec.policyTypes possibly undefined` (despite being explicitly defined above but it seems like typescript doesn't fully check for non-nullity of nested types?)

My first intent was to write `res.spec.policyTypes!.push(networkPolicyTypeIngress);` which passes the TS compilation however it triggers a linting error later (`!` is apparently forbidden), so I preferred to rewrite rather than ignoring the linting error.

Reply:
Ah thanks for the information. There isn't anything innately wrong with what you did here... my only concern was if later this `const policyTypes` changes to `let policyTypes` you may get yourself into the loss of an object reference and break the updating logic.

It's probably highly unlikely this will happen so I suppose we can go with this change. This file is somewhat concerning with this reliance on object mutation, but that's a different topic for a different PR/issue.

Reply:
To be honest I'd be in favour of removing the linter rule that forbids `!`, at some point; although the intent is good, but the TS flow control being not perfect, in practice it's common to see cases where it's counter-productive, like here or in some constructs with callbacks.